--- ./build/bext/opennurbs/CMakeLists.txt	2024-11-15 03:26:25.732356163 -0500
+++ ./build/bext/opennurbs/CMakeLists.txt_patched	2024-11-15 06:05:05.615289651 -0500
@@ -7,6 +7,9 @@
 
 if (ENABLE_OPENNURBS)
 
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0 -g")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0 -g")
+
   git_submodule_init(opennurbs opennurbs.h)
 
   RegisterDeps(OPENNURBS)
@@ -48,6 +51,8 @@
     $<$<BOOL:${ZLIB_TARGET}>:-DZ_PREFIX_STR=${Z_PREFIX_STR}>
     -DZLIB_ROOT=$<$<BOOL:${ZLIB_TARGET}>:${CMAKE_BUNDLE_INSTALL_PREFIX}>
     -DBUILD_STATIC_LIBS=${BUILD_STATIC_LIBS}
+    -DCMAKE_CXX_FLAGS="-O0 -g"
+    -DCMAKE_C_FLAGS="-O0 -g"
     -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
     -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
     -DCMAKE_INSTALL_PREFIX=${CMAKE_BUNDLE_INSTALL_PREFIX}
